source/core/render/trace.cpp

```diff
@@ -339,12 +339,13 @@ bool Trace::FindIntersection(ObjectPtr object, Intersection& isect, const Ray& r
         BBoxVector3d invdir;
         BBoxDirection variant;
 
-        Vector3d tmp(1.0 / ray.GetDirection()[X], 1.0 / ray.GetDirection()[Y], 1.0 /ray.GetDirection()[Z]);
+        Vector3d tmp(1.0 / ray.GetDirection()[X], 1.0 / ray.GetDirection()[Y], 1.0 / ray.GetDirection()[Z]);
         origin = BBoxVector3d(ray.Origin);
         invdir = BBoxVector3d(tmp);
         variant = (BBoxDirection)((int(invdir[X] < 0.0) << 2) | (int(invdir[Y] < 0.0) << 1) | int(invdir[Z] < 0.0));
 
-        if(object->Intersect_BBox(variant, origin, invdir, closest) == false)
+        // Note: At leaf when called. Suspect not strictly needed. However, without slightly slower in performance.
+        if((object->Intersect_BBox(variant, origin, invdir, closest) == false) && (sceneData->boundingMethod != 0))
```

Comment:
Hmm. Functionally the same so I have no objection. However, I intentionally ordered it with the bounding method test second because the test in normal use(1) will be true and unimportant when the bbox test is false. In other words, in normal use both the bounding method test and the bbox test will have to be done, if the method test is first, while ordered as it is, a lazy evaluation is possible when the bbox test is false. 

Do you still want to change the order? 

(1) I see the -mb mode as useful while debugging issues, but very useful otherwise.